util/qc_executor.py

```diff
@@ -98,48 +90,65 @@ def qc_check(self, parameter, dataset):
             if 'strict_validation' not in qcs[function_name]:
                 local_qc_args[function_name]['strict_validation'] = False
 
-            qc_process = None
-            try:
-                # call qc function in a separate process to deal with crashes, e.g. segfaults
-                module = importlib.import_module(ParameterFunction.query.filter_by(function=function_name)
-                                                 .first().owner)
-                parent_conn, child_conn = Pipe()
-                qc_process = Process(target=self.qc_process_wrapper, args=(child_conn, getattr(module, function_name),
-                                                                           local_qc_args.get(function_name)))
-                qc_process.start()
-                # close unused end of duplex pipe
-                child_conn.close()
-                results = parent_conn.recv()
-                qc_process.join()
-
-                # if qc function threw an exception, log it and try the next function
-                if isinstance(results, Exception):
-                    log.error('<%s> Failed to execute QC %s \n%s', self.request_id, function_name, results)
+            module = importlib.import_module(ParameterFunction.query.filter_by(function=function_name)
+                                             .first().owner)
+            # call qc function in a separate process to deal with crashes, e.g. segfaults
+            r, w = os.pipe()
+            processid = os.fork()
+            if processid == 0:
+                # child process
+                try:
+                    os.close(r)
+                    w = os.fdopen(w, 'w')
```

Comment:
To satisfy my own curiosity, I tried open instead of os.fdopen, but it led to errors. Using 'with os.fdopen(...) as w/r' clauses to handle opening and closing file descriptors. I discovered that if w.close() is not called in the finally clause, w.flush() must be called -- otherwise the parent process will not read anything for results_string from the pipe.